cnd/src/swap_protocols/rfc003/create_swap.rs

```diff
@@ -20,136 +20,45 @@ use genawaiter::{
     sync::{Co, Gen},
     GeneratorState,
 };
+use std::sync::Arc;
 
 /// Returns a future that tracks the swap negotiated from the given request and
-/// accept response on alpha ledger.
+/// accept response on a ledger.
 ///
 /// The current implementation is naive in the sense that it does not take into
 /// account situations where it is clear that no more events will happen even
 /// though in theory, there could. For example:
-/// - alpha funded
-/// - alpha refunded
+/// - funded
+/// - refunded
 ///
 /// It is highly unlikely for Bob to fund the HTLC now, yet the current
 /// implementation is still waiting for that.
-pub async fn create_alpha_watcher<D, A, AI, BI>(
+pub async fn create_watcher<D, S, L, A, H, I, T>(
```

Comment:
We should leave some easter eggs in the codebase where we write funny things with these type parameters xD

Reply:
`<S, H, I, T, L, A, D>`?

Reply:
How long did you work on this @da-kami ?

Reply:
30sec